Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -229,22 +274,26 @@ private void RunWhileSuspended(IEnumerable<TestMethod> tests)
                                     Logger.Trace(trace, "Unexpected COMException when running TestInitialize");
                                     continue;
                                 }
+
+                                _uiDispatcher.FlushMessageQueue();
```

Comment:
This seems to be the only one place in the `RunWhileSuspended` method where we pump the message but we've handled several other events. Shouldn't we pump before/after those other events?

Reply:
That was intentional.  I'm pumping here to catch a possible cancel button press, and by my read that was the only safe place to do it given the requirements for running the test setup and teardown procedures. The templates are probably safe to cancel in the middle of those, but we run every setup method we find so I'm thinking it's best to treat those as a unit (and not to cancel those steps). The rest of the pumping is done prior to raising status events from the `TestEngine`.  In those instances, we _know_ the apartment boundary is in a good state to pump.

Reply:
Then that needs to be put in a comment (xmldoc?). That intention is not apparent from reading the source code alone and we'll need to know that when we modify this in future.

Reply:
OK, I'll actually comment _all_ uses of `FlushMessageQueue` to be safe. There's already an extensive xmldoc warning on the interface itself about safe usage (you should see the tooltip).